dev/NavigationView/NavigationView.cpp

```diff
@@ -1592,6 +1770,161 @@ void NavigationView::UpdateVisualStateForDisplayModeGroup(const winrt::Navigatio
     }
 }
 
+void NavigationView::OnNavigationViewItemTapped(const winrt::IInspectable& sender, const winrt::TappedRoutedEventArgs& args)
+{
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        if (IsSettingsItem(nvi))
+        {
+            OnSettingsInvoked();
+        }
+        else
+        {
+            OnNavigationViewItemInvoked(nvi);
+        }
+        nvi.Focus(winrt::FocusState::Pointer);
+    }
+}
+
+void NavigationView::OnNavigationViewItemKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args)
+{
+    auto key = args.Key();
+
+    if (auto nvi = sender.try_as<winrt::NavigationViewItem>())
+    {
+        if (IsSettingsItem(nvi))
+        {
+            // Because ListViewItem eats the events, we only get these keys on KeyDown.
+            if (key == winrt::VirtualKey::Space ||
+                key == winrt::VirtualKey::Enter)
```

Comment:
>Enter [](start = 42, length = 5)

Gamepad can't select the settings button? home and end keys don't do anything on the settings button?

Reply:
Yep, home and end keys don't do anything when on settings. Gamepad can select the settings button, but it comes from KeyUp. I did a minor refactor here, so should make more sense now.